script/install-local-registry.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+CONTROL_PLANE_CONTAINER="kubeapps-ci-control-plane"
+REGISTRY_NS=ci
+DOCKER_REGISTRY_HOST=local-docker-registry
+DOCKER_REGISTRY_PORT=5600
+DOCKER_REGISTRY_URL="https://$DOCKER_REGISTRY_HOST:$DOCKER_REGISTRY_PORT"
+DOCKER_USERNAME=testuser
+DOCKER_PASSWORD=testpassword
+
+installLocalRegistry() {
+    PROJECT_PATH=$1
+
+    kubectl create namespace $REGISTRY_NS
+
+    # Add our CA to the node
+    docker cp "$(mkcert -CAROOT)"/rootCA.pem "$CONTROL_PLANE_CONTAINER:/usr/share/ca-certificates/local-rootCA.pem"
+    docker exec --user root $CONTROL_PLANE_CONTAINER sh -c "echo 'local-rootCA.pem' >> /etc/ca-certificates.conf"
+    docker exec --user root $CONTROL_PLANE_CONTAINER update-ca-certificates -f
+
+    # Restart containerd to make the CA addition effective
+    docker exec --user root $CONTROL_PLANE_CONTAINER systemctl restart containerd
```

Comment:
Key step in order to get CA certificates refreshed for Kubelet.

Reply:
Nice - I bet that one line cost a bit... sometimes I wish it were possible to show how much one line costs after feeling like I've been banging my head against a wall :)

Reply:
Woah, that's totally true... it seems a tricky one.

I thought this change in the CA could be done as a kubeadm config yaml during the cluster creation in kind (below), but you'll probably have explored many paths and this one is the simplest one for our use case, I guess, so +1 and thanks for coming up with it :P 

````
apiVersion: kubeadm.k8s.io/v1beta3
kind: ClusterConfiguration
controllerManager:
  extraArgs:
    cluster-signing-cert-file: /etc/kubernetes/pki/ca.crt
    cluster-signing-key-file: /etc/kubernetes/pki/ca.key
````

Reply:
| I bet that one line cost a bit.

Finding that out unlocked the whole thing to me after banging my head against the wall.

| you'll probably have explored many paths

Indeed, tried many ways to get it working, including the use of `/etc/kubernetes/pki/ca*` for generating the new certificate. Didn't work.
It seemed to me that the issue is when the "kubelet" establishes a TLS connection to contact the image repository. Acting as a client, it uses a set of CAs available form the underlying OS in which the `/etc/kubernetes/pki/ca.crt`, for some reason, is not included. Therefore, I needed to add a new CA to sign the certificate.